sdk/nodejs/output.ts

```diff
@@ -128,6 +130,20 @@ class OutputImpl<T> implements OutputInstance<T> {
             promise: Promise<T>,
             isKnown: Promise<boolean>,
             isSecret: Promise<boolean>) {
+
+        const value = Promise.all([isKnown, promise]).then(([known, val]) => (withUnknowns?: boolean) => {
+            // If the caller did not explicitly ask to see unknown values, simply return val.
+            if (!withUnknowns) {
+                return containsUnknowns(val) ? <T><any>undefined : val;
+            }
+
+            return val;
+        });
```

Comment:
oh wow... this is returning a Promise<Function>  i didn't even get that at first.